src/components/Settings/index.js

```diff
@@ -1,10 +1,113 @@
 import React, { Component } from 'react'
+import { reduxForm, Field, FieldArray } from 'redux-form'
+import Input from 'components/FormInput'
+
+
+class Settings extends Component {
+
+  renderForm() {
+    return (
+      <form>
+        <div className="row">
+          <div className="col-md-12">
+            <FieldArray name="settings" component={renderSettings} />
+          </div>
+        </div>
+      </form>
+    )
+  }
 
-export default class Settings extends Component {
   render() {
     return (
-      <div className="settingsPage">
+      <div className="container">
+        {this.renderForm()}
       </div>
     )
   }
 }
+
+const renderSettings = ({ fields, meta: { error } }) =>
+      <div className="row">
+        <div className="col-md-12">
+          <button type="button" onClick={() => fields.push({key: '', value: ''})} className="btn">
+            {fields.getAll() && fields.getAll().length > 0 ? 'ADD ANOTHER' : 'ADD'}
+          </button>
+        <div>
+      </div>
+      {fields.map((item, index) =>
+        <div className="row" key={index}>
+          <div className="col-md-12">
+            <h4>
+              Setting #{index + 1}
+            </h4>
+            <Field
+              name={`${item}.name`}
+              type="text"
+              component={Input}
+              label="NAME"
+              placeholder="George"
+            />
+            <Field
+              name={`${item}.address`}
+              type="text"
+              component={Input}
+              label="ADDRESS"
+              placeholder="0x12e87B8CE41184E0688027f370A972A436ABE34e"
+            />
+            <button
+              className="btn btn-primary"
+              type="button"
+              title="Remove"
+              onClick={() => fields.remove(index)}
+            >
+              REMOVE
+            </button>
+          </div>
+        </div>
+      )}
+      </div>
+    </div>
+
+// TODO move validate functions to a generic utilities file
+const validate = values => {
+  const errors = {}
+  const settingsErrors = []
+  for (var index=0; index<values.settings.length; index++) {
+    const error = {}
+    const item = values.settings[index]
+    // Name errors
+    if (!item.name || item.name.length < 3) {
+      error.name = 'Required'
+      settingsErrors[index] = error
+    }
+    const occurrences = values.settings.filter((other) => {return item.address == other.address})
```

Comment:
This can be written as 
`(other) => item.address == other.address`

Should be a linter error I think